test/y2add_on/clients/add-on_auto_test.rb

```diff
@@ -36,53 +256,73 @@
         ]
       end
 
-      context "and there are add-ons products" do
-        let(:add_on_products) do
-          [
-            {
-              "alias"       => "produc_alias",
-              "media_url"   => "http://product.url",
-              "name"        => "updated_repo",
-              "priority"    => 20,
-              "product_dir" => "/"
-            }
-          ]
-        end
-        let(:repos_to_store) do
-          [
-            {
-              "SrcId"        => 1,
-              "autorefresh"  => true,
-              "enabled"      => true,
-              "keeppackaged" => false,
-              "name"         => "updated_repo",
-              "priority"     => 20,
-              "service"      => ""
-            },
-            {
-              "SrcId"        => 2,
-              "autorefresh"  => true,
-              "enabled"      => true,
-              "keeppackaged" => false,
-              "name"         => "untouched_repo",
-              "priority"     => 99,
-              "service"      => ""
-            }
-          ]
-        end
+      before do
+        allow(Yast::AddOnProduct).to receive(:add_on_products).and_return(add_on_products)
+        allow(Yast::Pkg).to receive(:SourceEditSet)
+        allow(Yast::Pkg).to receive(:SourceCreate).and_return(1)
+        allow(Yast::Pkg).to receive(:SourceEditGet).and_return(repos)
+      end
 
+      # FIXME: improve that WIP scenarios/contexts
+      context "and product creation fails" do
         before do
-          allow(Yast::AddOnProduct).to receive(:add_on_products).and_return(add_on_products)
-          allow(Yast::Pkg).to receive(:SourceEditSet)
-          allow(Yast::Pkg).to receive(:SourceCreate).and_return(1)
-          allow(Yast::Pkg).to receive(:SourceEditGet).and_return(repos)
+          allow(Yast::Pkg).to receive(:SourceCreate).and_return(-1)
         end
 
-        it "stores repos according to information given" do
-          expect(Yast::Pkg).to receive(:SourceEditSet).with(repos_to_store)
+        context "ask_on_error=true" do
+          let(:ask_on_error) { true }
 
-          subject.main
+          it "ask to make it available" do
+            expect(Yast::Popup).to receive(:ContinueCancel)
+
+            subject.write
+          end
         end
+
+        context "ask_on_error=false" do
+          before do
+            allow(Yast::Popup).to receive(:ContinueCancel).and_return(false)
+          end
+
+          let(:ask_on_error) { false }
+
+          it "report error" do
+            expect(Yast::Report).to receive(:Error)
+
+            subject.write
+          end
+        end
+      end
+
+      it "stores repos according to information given" do
+        expect(Yast::Pkg).to receive(:SourceEditSet).with(repos_to_store)
+
+        subject.write
+      end
+    end
+  end
+
+  describe "#read" do
+    context "when cannot lock package" do
```

Comment:
The module name is a bit confusing, but it's not a package lock but a *package manager lock* (so only one application can use it at one time).